### PR TITLE
Setup electron store

### PR DESCRIPTION
**Pull request by dnlnrs**
References #52 
- Added NeDB as base database (https://github.com/seald/nedb/)
- Added logic to expose electron API through ipc messages: settings are passed from the worker to the renderer only using a specific "path" and exposed functions (without exposing node to the webview)
- Removed localStorage logic

### CODE REVIEW

**Comment by MystPi** [comment]
Why exactly do we need this? localStorage works great for this situation.

**Comment by dnlnrs** [comment]
We can use the same logic to store history and bookmarks and in this way the config can be backupped (it's just a file).

**Comment by MystPi** [comment]
Okay, this seems pretty good! You'll have to pull the latest changes from main, though, as I split up `browser.js` into four files. Sorry for the inconvenience.

**Comment by MystPi** [comment]
You should probably fork this from v2.2.4 as I have fixed a few bugs in it since you made this PR.

**Comment by MystPi** [comment]
How come most of `package-lock.json` is deleted?

**Comment by dnlnrs** [comment]
Good question!
I'll re-update and check why

**Comment by MystPi** [comment]
Also, can I merge this into v2.2.4 instead of main? I've been meaning to write some contributing guidelines about it but I never got around to it.

**Comment by dnlnrs** [comment]
Yes I think you can merge it in v2.2.4

**Comment by MystPi** [comment]
I'll merge once the `package-lock.json` is fixed. :)

**Comment by dnlnrs** [comment]
> I'll merge once the `package-lock.json` is fixed. :)

Should be ok now!

**Comment by MystPi** [comment]
> > I'll merge once the `package-lock.json` is fixed. :)
> 
> Should be ok now!

Great!

**Comment by MystPi** [comment]
Now it's always giving me this error:
Screenshot 2021-12-19 12 30 44 PM

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+    }, { upsert: true }, (err, numReplaced, upsert) => {
+      // TODO: handle errors
+    })
+  }
+})
+
+ipcMain.handle('load-user-config', async (event, arg) => {
+  if (typeof arg === 'string') {
+    return new Promise((resolve, reject) => {
+      userConfigDb.findOne({ key: arg }, (err, { value }) => {
```
Comment (MystPi) [inline src/index.js:27]: The error is on this line.

Comment (MystPi) [reply src/index.js:27]: Maybe it's because .config/user-config.db doesn't exist and doesn't contain the required keys?

**Comment by MystPi** [comment]
I should have tested it first. I'm going to revert it for now, but when the issues are fixed I'll re-merge.

**Comment by dnlnrs** [comment]
> 

I've just release a PR to fix it

**Pull request merged**